trio/_subprocess.py

```diff
@@ -377,13 +378,42 @@ async def open_process(
     return Process._create(popen, trio_stdin, trio_stdout, trio_stderr)
 
 
+async def _windows_deliver_cancel(p):
+    try:
+        p.terminate()
+    except OSError as exc:
+        warnings.warn(
+            RuntimeWarning(f"TerminateProcess on {p!r} failed with: {exc!r}")
+        )
+
+
+async def _posix_deliver_cancel(p):
+    try:
+        p.terminate()
+        await trio.sleep(5)
+        warnings.warn(
+            RuntimeWarning(
+                f"process {p!r} ignored SIGTERM for 5 seconds. "
+                f"(Maybe you should custom deliver_cancel?) Trying SIGKILL."
```

Comment:
The grammar nitpicker in me wants this to be "customize deliver_cancel" or "specify a custom deliver_cancel".

Reply:
Whoops, that was an editing fail, yeah. #1531